packages/@aws-cdk/aws-synthetics/test/canary.test.ts

```diff
@@ -0,0 +1,251 @@
+import '@aws-cdk/assert/jest';
+import { arrayWith, objectLike } from '@aws-cdk/assert';
+import * as iam from '@aws-cdk/aws-iam';
+import * as s3 from '@aws-cdk/aws-s3';
+import { App, Duration, Stack } from '@aws-cdk/core';
+import * as synthetics from '../lib';
+
+let stack: Stack;
+beforeEach(() => {
+  stack = new Stack(new App(), 'canaries');
+});
+
+test('Create a basic canary', () => {
+  // WHEN
+  new synthetics.Canary(stack, 'Canary', {});
+
+  // THEN
+  expect(stack).toHaveResourceLike('AWS::Synthetics::Canary', {
+    Name: 'canariescanary8f7842',
+    Code: {
+      Handler: 'index.handler',
+      Script: 'exports.handler = async () => {\nconsole.log(\'hello world\');\n};',
+    },
+    RuntimeVersion: 'syn-1.0',
+  });
+});
+
+test('Canary can have specified name', () => {
+  // WHEN
+  new synthetics.Canary(stack, 'Canary', {
+    canaryName: 'mycanary',
+  });
+
+  // THEN
+  expect(stack).toHaveResourceLike('AWS::Synthetics::Canary', {
+    Name: 'mycanary',
+    Code: {
+      Handler: 'index.handler',
+      Script: 'exports.handler = async () => {\nconsole.log(\'hello world\');\n};',
+    },
+    RuntimeVersion: 'syn-1.0',
+  });
+});
+
+test('Canary returns error when name is specified incorrectly', () => {
+  expect(() => new synthetics.Canary(stack, 'Canary', {canaryName: 'myCanary'})).toThrowError('Canary Name must be lowercase, numbers, hyphens, or underscores (no spaces)');
+});
+
+test('Canary name must be less than 21 characters', () => {
+  expect(() => new synthetics.Canary(stack, 'Canary', {canaryName: 'canary-name-super-long'})).toThrowError('Canary Name must be less than 21 characters');
+});
+
+test('Canary can have specified IAM role', () => {
+  // GIVEN
+  const role = new iam.Role(stack, 'role', {
+    assumedBy: new iam.ServicePrincipal('lambda.amazonaws.com'),
+  });
+
+  role.addToPolicy(new iam.PolicyStatement({
+    resources: ['*'],
+    actions: [
+      's3:PutObject',
+      's3:GetBucketLocation',
+      's3:ListAllMyBuckets',
+      'cloudwatch:PutMetricData',
+      'logs:CreateLogGroup',
+      'logs:CreateLogStream',
+      'logs:PutLogEvents',
+    ],
+  }));
```

Comment:
I doubt you're validating this (because you can't), so you probably should just elude that out.

```suggestion
  // role.addToPolicy(/* required permissions per the documentation */);
```